test/jdk/java/lang/management/ThreadMXBean/Locks.java

```diff
@@ -445,11 +460,18 @@ private static void doCheck(ThreadInfo[] infos, String lock, long[] expectedThre
                     "Can't retrieve ThreadInfo for the blocked thread");
         }
 
+        // Follow chain of locks:
         long[] threads = new long[10];
         int count = 0;
         threads[count++] = ownerInfo.getThreadId();
         while (ownerInfo.getThreadState() == Thread.State.BLOCKED) {
             ownerInfo = findOwnerInfo(infos, lock);
+            log("ownerInfo = %s", ownerInfo);
+            if (ownerInfo.getThreadName().contains("ForkJoinPool")) {
+                log ("skipping %s", ownerInfo);
+                lock = ownerInfo.getLockName();
+                continue;
+            }
```

Comment:
What happens if you don't exclude these ForkJoinPool threads? What about the myriad of other threads that the JVM always starts up. Should they not also be skipped if you are going to skip ForkJoinPool threads?

Reply:
Thanks for the feedback, updated.
On this point, the method is only invoked on a very limited set of locks and threads.  It is called to examine LockAThread and LockBThread, and the test main thread (if not virtual).  So it will not (should not!) see a lock owner which is some other thread, it's not a useful generic utility method.

Reply:
This is not that easy to make this conclusion.
I'm thinking if placing a comment would help.

Reply:
Sure.  I checked again and without this "skip ForkJoinPool" logic,  JTREG_TEST_THREAD_FACTORY=Virtual causes failures.

This method checks in all the threadInfos, that the specified lock is blocking the expectedThreads (our test threads t1, t2 and mainThread if it's not Virtual)

If it sees a FJP thread that is blocked on a java.lang.VirtualThread, and that's not one of the expected threads, so the counting is wrong and we fail.

I cannot say I like the loop. 8-)